domain/src/main/java/me/androidbox/domain/authentication/usecase/RegisterUseCase.kt

```diff
@@ -0,0 +1,5 @@
+package me.androidbox.domain.authentication.usecase
+
+fun interface RegisterUseCase {
+    suspend fun execute(fullName: String, email: String, password: String)
+}
```

Comment:
There's no real benefit in having a usecase as an interface
Since you want to actually test the functionality of said usecase

Reply:
@MKiperszmid the implementation is called `RegisterUseCaseImp`

Reply:
Gotcha

I was saying, since usually you tend to want the usecase to be ran on the test also

Since that's how you test it's actually working

Reply:
Yeah, have to agree with Martin here. You never want to replace the implementation of a use case for your tests. They contain your app's business logic and already are isolated as much as possible. You only want to create fakes for your boundaries (so basically things that leave the context of your local app's environment such as remote APIs, local DBs, dealing with Android OS services like AlarmManager, etc.). You use none of these in a use case.

Abstractions are expensive, every abstraction you add, adds quite a lot of complexity and cognitive load to your project. General rule of thumb: If you'll only ever need one implementation of an abstraction, leave it away, it's not needed.